### PR TITLE
[helm] Fix release namespace detection and add helm_version to metric

### PR DESCRIPTION
**Pull request by yalosev**
Signed-off-by: Yuriy Losev <yuriy.losev@flant.com>

## Description
Fix release namespace detection.
Add helm_version to the metric.
Add 1.24 k8s unsupported APIs.

## Why do we need it, and what problem does it solve?
Helm release namespace unmarshalling from resource sometimes gives wierd results. Getting namespace from secret/cm is more predictable.
Adding `helm_version` value to metrics makes the process of figuring out a deprecated release easier.

## Changelog entries

```changes
section: helm
type: fix 
summary: Fix namespace detection for helm releases with unsupported/deprecated resources.
```

### CODE REVIEW

File: modules/013-helm/hooks/metrics.go [hunk 1]
```diff
@@ -285,7 +293,10 @@ func runReleaseProcessor(k8sCurrentVersion *semver.Version, input *go_hook.HookI
 type release struct {
 	Name      string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name,proto3"`
 	Namespace string `json:"namespace,omitempty" protobuf:"bytes,8,opt,name=namespace,proto3"`
-	Manifest  string `json:"manifest,omitempty" protobuf:"bytes,5,opt,name=manifest,proto3" `
+	Manifest  string `json:"manifest,omitempty" protobuf:"bytes,5,opt,name=manifest,proto3"`
+
+	// set helm version manually
+	HelmVersion string
```
Comment (nabokihms) [inline modules/013-helm/hooks/metrics.go:299]:
```suggestion
	HelmVersion string `json:"-"`
```

File: modules/013-helm/hooks/metrics.go [hunk 2]
```diff
@@ -193,6 +196,8 @@ func getHelm3Releases(ctx context.Context, client k8s.Client, releasesC chan<- *
 			if err != nil {
 				return 0, err
 			}
+			release.Namespace = secret.Namespace
```
Comment (nabokihms) [inline modules/013-helm/hooks/metrics.go:199]: Let's add a comment on what we did there and why.

**Comment by nabokihms** [comment]
I assume we also should add a test for it.

**Comment by z9r5** [comment]
Cherry-picked into release-1.33

**Pull request merged**